Frameworks/AddressBook/ABAddressBook.mm

```diff
@@ -16,44 +16,87 @@
 
 const CFStringRef ABAddressBookErrorDomain = static_cast<const CFStringRef>(@"ABAddressBookErrorDomain");
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 ABAddressBookRef ABAddressBookCreate() {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return ABAddressBookCreateWithOptions(NULL, NULL);
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 ABAddressBookRef ABAddressBookCreateWithOptions(CFDictionaryRef options, CFErrorRef* error) {
-    UNIMPLEMENTED();
-    return StubReturn();
+    _ABAddressBook* addressBook = [[_ABAddressBook alloc] init];
+    if (addressBook.contactStore == nil) {
+        // There was an error setting up the backing
+        // contact store, so just return NULL to signify
+        // the error.
+        if (error != NULL) {
+			// In this case, the caller is responsible to release error.
+            *error = CFErrorCreate(NULL, ABAddressBookErrorDomain, kABOperationNotPermittedByUserError, NULL);
+        }
+        return NULL;
+    } else {
+        return (__bridge_retained ABAddressBookRef)addressBook;
+    }
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 ABAuthorizationStatus ABAddressBookGetAuthorizationStatus() {
-    UNIMPLEMENTED();
-    return kABAuthorizationStatusNotDetermined;
+    WFGUID* guid = [WFGUID guidWithGUID:_ABAddressBookContactsGUID];
+    WDEDeviceAccessInformation* deviceAccessInformation = [WDEDeviceAccessInformation createFromDeviceClassId:guid];
+    WDEDeviceAccessStatus currentStatus = deviceAccessInformation.currentStatus;
+
+    if (currentStatus == WDEDeviceAccessStatusAllowed) {
+        return kABAuthorizationStatusAuthorized;
+    } else if (currentStatus == WDEDeviceAccessStatusDeniedBySystem) {
+        return kABAuthorizationStatusRestricted;
+    } else if (currentStatus == WDEDeviceAccessStatusDeniedByUser) {
+        return kABAuthorizationStatusDenied;
+    } else { // WDEDeviceAccessStatusUnspecified
+        return kABAuthorizationStatusNotDetermined;
+    }
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Caveat
+ @Notes Requesting access should be done by declaring Contacts capabilities in the app's package manifest -- it cannot be granted at
+ runtime.
 */
 void ABAddressBookRequestAccessWithCompletion(ABAddressBookRef addressBook, ABAddressBookRequestAccessCompletionHandler completion) {
-    UNIMPLEMENTED();
+    CFErrorRef error = NULL;
+    bool accessGranted = ABAddressBookGetAuthorizationStatus() == kABAuthorizationStatusAuthorized;
+    if (!accessGranted) {
+        error = CFErrorCreate(NULL, ABAddressBookErrorDomain, kABOperationNotPermittedByUserError, NULL);
+    }
+
+    // Since we can't request access at runtime (Contacts capabilities must be declared
+    // in the app's package manifest), immediately call the completion handler with the current
+    // authorization status.
+    completion(accessGranted, error);
```

Comment:
>completion(accessGranted, error); [](start = 4, length = 33)

Maybe for compatibility with the reference platform we should dispatch this async.